Automattic/simplenote-macos
# Interlink References

### Details:
- Rebuilds the Metrics UI, based on NSTableView
- Implements new NSTableViewCell subclasses: `[MetricTableViewCell, ReferenceTableViewCell, SeparatorTableViewCell]`
- Metrics: We're now displaying inbound note references

@eshurakov Mind taking a look at this one Evgeny?
Thank youuu!

Ref. #663 

### Test: Multiple Selection
1. Log into your account!
2. Right click over **Note A** in your notes list: click over `Copy Internal Link`
3. Click over **Note B** and paste the contents of your clipboard
4. CMD + Click over **Note A** + **Note B** in your list
5. Click over the top right `( i )` icon

- [x] Verify only 5 rows show up:
- [x] Verify only `Words` and `Characters` are populated
- [x] Verify that editing either `Note A` or `Note B` in a secondary device (simplenote.com on the web works!) causes the UI to refresh


### Test: Single Selection
1. Log into your account!
2. Right click over **Note A** in your notes list: click over `Copy Internal Link`
3. Paste the contents of your clipboard on **at least** 4 notes (so that 4 notes reference back to Note A!)
4. Click over **Note A**
5. Click over the top right `( i )` icon

- [x] Verify that all of the inbound links (References!) you pasted in step 3 show up at the bottom of the Metrics UI
- [x] Verify that you can scroll **all of the contents** of the Metrics UI (every single row should move!)

### Release
These changes do not require release notes.


---

### Screenshots
<img width="300" alt="Screen Shot 2020-10-21 at 15 22 11" src="https://user-images.githubusercontent.com/1195260/96766432-7485d880-13b1-11eb-8327-b69ccafa78bd.png">

<img width="300" alt="Screen Shot 2020-10-21 at 15 22 24" src="https://user-images.githubusercontent.com/1195260/96766411-6df76100-13b1-11eb-8c9d-7049a6066278.png">
